### DISCUSSION TITLE
Custom controls

### DISCUSSION THREAD

**Discussion by gbernaldo82** [frang75/nappgui_src#140] [Q&A] [answered]
Hi, good morning!

I'm new in the discussion sections, so, please, bear with me if the questions has been asked before. I trieed to look for sth similar but I didn't find anything. So, my question is if there is any possibility to create custom controls and if so, how can I do it? Is there any section in the documentation (or maybe some example) about how to do it?

By custom controls I mean controls with custom behaviour (for example, a button which expands a panel when is pressed over it) instead of custom look and feel. 

Thank you so much in advance!

I've been testing the SDK and I think is vereything I need in my life :D

**Comment by frang75** [comment, marked as answer]
Hi @gbernaldo82! Thank you for use NAppGUI-SDK.

For create custom controls you have to use [View](https://nappgui.com/en/gui/view.html) component, that allows you to draw anything and get the mouse/keyboard events. You have several examples:
https://nappgui.com/en/howto/col2dhello.html
https://nappgui.com/en/demo/die.html
https://nappgui.com/en/demo/bode.html

You can also use the undocumented API `drawctrl.h` that allows you draw some items with the native look. e.g Headers and Checkboxes. Take a look to [ListBox](https://nappgui.com/en/gui/listbox.html) and [TableView](https://nappgui.com/en/gui/tableview.html). In NAppGUI these components are "custom controls" instead a wrapper over native components.

On the other hand, if what you are looking for are controls that change the morphology of the interface, take a look at these demos.

**Flyout Windows**
https://nappgui.com/en/howto/guihello.html#h11
https://github.com/frang75/nappgui_src/issues/50

**Dynamic Layouts**
https://nappgui.com/en/howto/guihello.html#h18

**Panel Multilayout**
https://nappgui.com/en/gui/panel.html#h1
![MultiLayout](https://github.com/frang75/nappgui_src/assets/42999199/01d6aae0-40cc-48d6-ae6e-43c8aa5ab7e2)

**Button Show/Hide Panel**
https://nappgui.com/en/demo/products.html#h4.2
![HidePanel](https://github.com/frang75/nappgui_src/assets/42999199/7b454a3c-6bb7-4a75-875c-da563dbc19a5)

If this doesn't solve your problem, open an issue indicating what exactly you need to see the best way to implement it.

**Reply by gbernaldo82** [reply]
Hi @frang75 

Thank you so much for your answer. I'll look at the code and I'll tell you sth as soon as I get more questions.
